src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -155,7 +157,12 @@ export class InstructorStudentListPageComponent implements OnInit {
             this.statusMessageService.showErrorToast(resp.error.message);
           });
         }, (resp: ErrorMessageOutput) => {
-          courseTab.hasLoadingFailed = true;
+          if (resp.error.message === 'You are not authorized to access this resource.') {
+            courseTab.isAbleToViewStudents = false;
```

Comment:
You can use http response status to differentiate the errors, to see what status codes are returned from the backend see `WebApiServlet.java`